iavl/iavl_path.go

```diff
@@ -0,0 +1,148 @@
+package iavl
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/tendermint/go-wire/data"
+)
+
+// PathToKey represents an inner path to a leaf node.
+// Note that the nodes are ordered such that the last one is closest
+// to the root of the tree.
+type PathToKey struct {
+	LeafHash   data.Bytes           `json:"leaf_hash"`
+	InnerNodes []IAVLProofInnerNode `json:"inner_nodes"`
+}
+
+func (p *PathToKey) String() string {
+	str := ""
+	for i := len(p.InnerNodes) - 1; i >= 0; i-- {
+		str += p.InnerNodes[i].String() + "\n"
+	}
+	str += fmt.Sprintf("hash(leaf)=%s\n", p.LeafHash)
+	return str
+}
+
+func (p *PathToKey) verify(leafNode IAVLProofLeafNode, root []byte) error {
```

Comment:
Could we add a comment to verify? I find myself having to scroll back to its code body to read through the entire definition.

Perhaps?
```go
// verify firstly checks that the leafNode's hash matches the path's
// LeafHash and that root is the merkle hash of all the inner nodes.